app/components/ChartSettings/modules/XAxis.js

```diff
@@ -7,31 +8,71 @@ import {
 } from '../../../constants';
 import { getObjArrayKey } from '../../../utils/misc';
 
-export default function XAxis(props) {
+function XAxis({ defaultExpand, options, dataFormat }) {
+  const handleXDomain = ({ name }, value) => {
```

Comment:
You could abstract this, since it's a dupe of handleXDomain. The argument to differentiate would be the path of the object that holds the value of the domain.